### PR TITLE
input.cc: close dialog when parent is hidden

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

File: selfdrive/ui/qt/widgets/input.cc [hunk 1]
```diff
@@ -5,7 +5,20 @@
...
-InputDialog::InputDialog(const QString &prompt_text, QWidget *parent) : QDialog(parent) {
+QDialogBase::QDialogBase(QWidget *parent) : QDialog(parent) {
+  Q_ASSERT(parent != nullptr);
+  parent->installEventFilter(this);
+}
+
+bool QDialogBase::eventFilter(QObject *o, QEvent *e) {
+  if (o == parent() && e->type() == QEvent::Hide) {
+    close();
+    return false;
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/widgets/input.cc:16]: do we need to return here?

File: selfdrive/ui/qt/widgets/input.cc [hunk 2]
```diff
@@ -5,7 +5,20 @@
...
 
-InputDialog::InputDialog(const QString &prompt_text, QWidget *parent) : QDialog(parent) {
+QDialogBase::QDialogBase(QWidget *parent) : QDialog(parent) {
+  Q_ASSERT(parent != nullptr);
+  parent->installEventFilter(this);
+}
+
+bool QDialogBase::eventFilter(QObject *o, QEvent *e) {
+  if (o == parent() && e->type() == QEvent::Hide) {
+    close();
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/widgets/input.cc:15]: should we reject instead?

File: selfdrive/ui/qt/widgets/ssh_keys.cc [hunk 3]
```diff
@@ -26,7 +26,7 @@ SshControl::SshControl() : AbstractControl("SSH Keys", "Warning: This grants SSH
 
   QObject::connect(&btn, &QPushButton::released, [=]() {
     if (btn.text() == "ADD") {
-      QString username = InputDialog::getText("Enter your GitHub username");
+      QString username = InputDialog::getText("Enter your GitHub username", 39, this);
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/widgets/ssh_keys.cc:29]: min length added by mistake?

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]
nice cleanup

File: selfdrive/ui/qt/widgets/ssh_keys.cc (hunk 3 above)
Comment (deanlee) [reply selfdrive/ui/qt/widgets/ssh_keys.cc:29]: github limits the username to 39 characters.

Comment (pd0wm) [reply selfdrive/ui/qt/widgets/ssh_keys.cc:15]: This is a minimum length (e.g. for wifi passwords).

Comment (deanlee) [reply selfdrive/ui/qt/widgets/ssh_keys.cc:15]: ....my bad

**Comment by adeebshihadeh** [comment]
Nice, I really like this.

**Pull request merged**